index.js

```diff
@@ -2,54 +2,48 @@
 var bodyParser = require('body-parser');
 var express = require('express');
 var gitHubWebHook = require('express-github-webhook');
-var nconf = require('nconf');
 
 var commentOnClosedIssue = require('./lib/commentOnClosedIssue');
+var Settings = require('./lib/Settings');
 
 var app = express();
 module.exports = app;
 
-nconf.env('__')
-    .file({
-        file: 'config.json'
-    });
-
-nconf.defaults({
-    port: 5000,
-    secret: '', // Repository secret to verify incoming WebHook requests from GitHub
-    gitHubToken: '', // Token used to verify outgoing requests to GitHub repository
-    listenPath: '/' // Path on which to listen for incoming requests
-});
+try {
```

Comment:
Make this return a promise and handle the rejection rather than catching an exception.  Also, make the function fail gracefully with an error instead of just throwing.